.github/workflows/test-build-deploy.yml

```diff
@@ -338,17 +338,34 @@ jobs:
         uses: actions/download-artifact@v4
         with:
           name: Race-enabled Mimir
-      - name: Load race-enabled mimir into Docker
```

Comment:
Why move this? It made sense to keep it together with `Download Archive with Docker Images` step.

Reply:
because download archive with docker images step doesn't depends on the input tag, i wanted to group whatever needs distroless tag together to make it more evident.